### ISSUE TITLE
A nice to have: automatically activate instances that are known to the container

### ISSUE THREAD

**Issue by ghost** [LeeCheneler/EasyIoC#5]
Assume that the container knows/has registered how to create `IClassA` (concrete implementation being `ClassA`).

Also assume that I have `IClassB`, with the concrete implementation being `ClassB`.

Assume my concrete implementation of `ClassB` takes a dependency of `IClassA`.

In order to register `IClassB` with the container, I have to explicitly register how to create `ClassB` - i.e. I have to supply a `Func{object}` which provides an instance of `ClassB`, which injects an instance of `ClassA` (which is created either through `IEasyServiceContainer.Activate` or through my own means).

It would be neat that, EasyIoC, upon seeing the dependency on `IClassA`, knows how to create the concrete implementation of `IClassA` (as it has been registered prior) doesn't need this `Func{object}`, and instead, behind the scenes calls `IEasyServiceContainer.Activate` to provide `ClassB` with it's instance of `ClassA`.

In a nutshell, it would be good if EasyIoC automatically activates the instance if it knows about it.
